sktree/stats/forestht.py

```diff
@@ -206,13 +206,16 @@ def _get_estimators_indices(self, sample_separate=False):
             indices_train, indices_test = train_test_split(
                 indices,
                 test_size=self.test_size,
+                stratify = y,
                 random_state=self._seeds,
+
             )
+            
             for _ in self.estimator_.estimators_:
                 yield indices_train, indices_test
 
-    @property
-    def train_test_samples_(self):
+    
+    def train_test_samples_(self,y):
```

Comment:
If y will be passed. There should be a test that y is of the same type during fitting. 

I.e. type of target, n samples, n outputs should be invariant.

Reply:
Okay, adding a check type now.